From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Driver Core: fix up all callers of class_device_create()

The previous patch adding the ability to nest struct class_device
changed the paramaters to the call class_device_create().  This patch
fixes up all in-kernel users of the function.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/i386/kernel/cpuid.c                |    2 +-
 arch/i386/kernel/msr.c                  |    2 +-
 drivers/block/aoe/aoechr.c              |    2 +-
 drivers/block/paride/pg.c               |    2 +-
 drivers/block/paride/pt.c               |    4 ++--
 drivers/char/dsp56k.c                   |    2 +-
 drivers/char/ftape/zftape/zftape-init.c |   12 ++++++------
 drivers/char/ip2main.c                  |   10 ++++++----
 drivers/char/ipmi/ipmi_devintf.c        |    2 +-
 drivers/char/istallion.c                |    3 ++-
 drivers/char/lp.c                       |    2 +-
 drivers/char/mem.c                      |    3 ++-
 drivers/char/misc.c                     |    2 +-
 drivers/char/ppdev.c                    |    2 +-
 drivers/char/raw.c                      |    4 ++--
 drivers/char/snsc.c                     |    2 +-
 drivers/char/stallion.c                 |    2 +-
 drivers/char/tipar.c                    |    2 +-
 drivers/char/tty_io.c                   |   10 +++++-----
 drivers/char/vc_screen.c                |    8 ++++----
 drivers/char/viotape.c                  |    4 ++--
 drivers/hwmon/hwmon.c                   |    2 +-
 drivers/ide/ide-tape.c                  |    4 ++--
 drivers/ieee1394/dv1394.c               |    2 +-
 drivers/ieee1394/raw1394.c              |    2 +-
 drivers/ieee1394/video1394.c            |    2 +-
 drivers/infiniband/core/ucm.c           |    2 +-
 drivers/input/evdev.c                   |    2 +-
 drivers/input/joydev.c                  |    2 +-
 drivers/input/mousedev.c                |    4 ++--
 drivers/input/tsdev.c                   |    2 +-
 drivers/isdn/capi/capi.c                |    2 +-
 drivers/macintosh/adb.c                 |    2 +-
 drivers/media/dvb/dvb-core/dvbdev.c     |    2 +-
 drivers/message/i2o/iop.c               |    2 +-
 drivers/mtd/mtdchar.c                   |    4 ++--
 drivers/net/ppp_generic.c               |    2 +-
 drivers/net/wan/cosa.c                  |    2 +-
 drivers/s390/char/tape_class.c          |    1 +
 drivers/s390/char/vmlogrdr.c            |    1 +
 drivers/scsi/ch.c                       |    2 +-
 drivers/scsi/osst.c                     |    2 +-
 drivers/scsi/sg.c                       |    2 +-
 drivers/scsi/st.c                       |    2 +-
 drivers/usb/core/devio.c                |    2 +-
 drivers/usb/core/file.c                 |    4 +++-
 drivers/usb/core/hcd.c                  |    3 ++-
 drivers/video/fbmem.c                   |    2 +-
 fs/coda/psdev.c                         |    4 ++--
 sound/core/sound.c                      |    2 +-
 sound/oss/soundcard.c                   |    4 ++--
 sound/sound_core.c                      |    5 +++--
 52 files changed, 84 insertions(+), 74 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/devio.c	2005-09-28 12:33:54.000000000 -0700
+++ gregkh-2.6/drivers/usb/core/devio.c	2005-09-28 12:40:33.000000000 -0700
@@ -1546,7 +1546,7 @@
 {
 	int minor = ((dev->bus->busnum-1) * 128) + (dev->devnum-1);
 
-	dev->class_dev = class_device_create(usb_device_class,
+	dev->class_dev = class_device_create(usb_device_class, NULL,
 				MKDEV(USB_DEVICE_MAJOR, minor), &dev->dev,
 				"usbdev%d.%d", dev->bus->busnum, dev->devnum);
 
--- gregkh-2.6.orig/drivers/usb/core/file.c	2005-09-28 12:33:54.000000000 -0700
+++ gregkh-2.6/drivers/usb/core/file.c	2005-09-28 12:40:33.000000000 -0700
@@ -164,7 +164,9 @@
 		++temp;
 	else
 		temp = name;
-	intf->class_dev = class_device_create(usb_class, MKDEV(USB_MAJOR, minor), &intf->dev, "%s", temp);
+	intf->class_dev = class_device_create(usb_class, NULL,
+					      MKDEV(USB_MAJOR, minor),
+					      &intf->dev, "%s", temp);
 	if (IS_ERR(intf->class_dev)) {
 		spin_lock (&minor_lock);
 		usb_minors[intf->minor] = NULL;
--- gregkh-2.6.orig/drivers/usb/core/hcd.c	2005-09-28 12:33:54.000000000 -0700
+++ gregkh-2.6/drivers/usb/core/hcd.c	2005-09-28 12:40:33.000000000 -0700
@@ -782,7 +782,8 @@
 		return -E2BIG;
 	}
 
-	bus->class_dev = class_device_create(usb_host_class, MKDEV(0,0), bus->controller, "usb_host%d", busnum);
+	bus->class_dev = class_device_create(usb_host_class, NULL, MKDEV(0,0),
+					     bus->controller, "usb_host%d", busnum);
 	if (IS_ERR(bus->class_dev)) {
 		clear_bit(busnum, busmap.busmap);
 		up(&usb_bus_list_lock);
--- gregkh-2.6.orig/drivers/video/fbmem.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/video/fbmem.c	2005-09-28 12:42:30.000000000 -0700
@@ -1030,7 +1030,7 @@
 			break;
 	fb_info->node = i;
 
-	fb_info->class_device = class_device_create(fb_class, MKDEV(FB_MAJOR, i),
+	fb_info->class_device = class_device_create(fb_class, NULL, MKDEV(FB_MAJOR, i),
 				    fb_info->device, "fb%d", i);
 	if (IS_ERR(fb_info->class_device)) {
 		/* Not fatal */
--- gregkh-2.6.orig/drivers/block/aoe/aoechr.c	2005-08-28 16:41:01.000000000 -0700
+++ gregkh-2.6/drivers/block/aoe/aoechr.c	2005-09-28 12:45:08.000000000 -0700
@@ -224,7 +224,7 @@
 		return PTR_ERR(aoe_class);
 	}
 	for (i = 0; i < ARRAY_SIZE(chardevs); ++i)
-		class_device_create(aoe_class,
+		class_device_create(aoe_class, NULL,
 					MKDEV(AOE_MAJOR, chardevs[i].minor),
 					NULL, chardevs[i].name);
 
--- gregkh-2.6.orig/drivers/block/paride/pg.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/block/paride/pg.c	2005-09-28 12:45:17.000000000 -0700
@@ -672,7 +672,7 @@
 	for (unit = 0; unit < PG_UNITS; unit++) {
 		struct pg *dev = &devices[unit];
 		if (dev->present)
-			class_device_create(pg_class, MKDEV(major, unit),
+			class_device_create(pg_class, NULL, MKDEV(major, unit),
 					NULL, "pg%u", unit);
 	}
 	err = 0;
--- gregkh-2.6.orig/drivers/block/paride/pt.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/block/paride/pt.c	2005-09-28 12:45:48.000000000 -0700
@@ -969,9 +969,9 @@
 
 	for (unit = 0; unit < PT_UNITS; unit++)
 		if (pt[unit].present) {
-			class_device_create(pt_class, MKDEV(major, unit),
+			class_device_create(pt_class, NULL, MKDEV(major, unit),
 					NULL, "pt%d", unit);
-			class_device_create(pt_class, MKDEV(major, unit + 128),
+			class_device_create(pt_class, NULL, MKDEV(major, unit + 128),
 					NULL, "pt%dn", unit);
 		}
 	goto out;
--- gregkh-2.6.orig/drivers/char/dsp56k.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/dsp56k.c	2005-09-28 12:47:17.000000000 -0700
@@ -514,7 +514,7 @@
 		err = PTR_ERR(dsp56k_class);
 		goto out_chrdev;
 	}
-	class_device_create(dsp56k_class, MKDEV(DSP56K_MAJOR, 0), NULL, "dsp56k");
+	class_device_create(dsp56k_class, NULL, MKDEV(DSP56K_MAJOR, 0), NULL, "dsp56k");
 
 	printk(banner);
 	goto out;
--- gregkh-2.6.orig/drivers/char/ip2main.c	2005-09-27 03:04:16.000000000 -0700
+++ gregkh-2.6/drivers/char/ip2main.c	2005-09-28 12:49:55.000000000 -0700
@@ -716,10 +716,12 @@
 			}
 
 			if ( NULL != ( pB = i2BoardPtrTable[i] ) ) {
-				class_device_create(ip2_class, MKDEV(IP2_IPL_MAJOR,
-						4 * i), NULL, "ipl%d", i);
-				class_device_create(ip2_class, MKDEV(IP2_IPL_MAJOR,
-						4 * i + 1), NULL, "stat%d", i);
+				class_device_create(ip2_class, NULL,
+						MKDEV(IP2_IPL_MAJOR, 4 * i),
+						NULL, "ipl%d", i);
+				class_device_create(ip2_class, NULL,
+						MKDEV(IP2_IPL_MAJOR, 4 * i + 1),
+						NULL, "stat%d", i);
 
 			    for ( box = 0; box < ABS_MAX_BOXES; ++box )
 			    {
--- gregkh-2.6.orig/drivers/char/istallion.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/istallion.c	2005-09-28 12:49:55.000000000 -0700
@@ -5238,7 +5238,8 @@
 
 	istallion_class = class_create(THIS_MODULE, "staliomem");
 	for (i = 0; i < 4; i++)
-		class_device_create(istallion_class, MKDEV(STL_SIOMEMMAJOR, i),
+		class_device_create(istallion_class, NULL,
+				MKDEV(STL_SIOMEMMAJOR, i),
 				NULL, "staliomem%d", i);
 
 /*
--- gregkh-2.6.orig/drivers/char/lp.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/lp.c	2005-09-28 12:48:20.000000000 -0700
@@ -804,7 +804,7 @@
 	if (reset)
 		lp_reset(nr);
 
-	class_device_create(lp_class, MKDEV(LP_MAJOR, nr), NULL,
+	class_device_create(lp_class, NULL, MKDEV(LP_MAJOR, nr), NULL,
 				"lp%d", nr);
 
 	printk(KERN_INFO "lp%d: using %s (%s).\n", nr, port->name, 
--- gregkh-2.6.orig/drivers/char/mem.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/mem.c	2005-09-28 12:49:55.000000000 -0700
@@ -919,7 +919,8 @@
 
 	mem_class = class_create(THIS_MODULE, "mem");
 	for (i = 0; i < ARRAY_SIZE(devlist); i++)
-		class_device_create(mem_class, MKDEV(MEM_MAJOR, devlist[i].minor),
+		class_device_create(mem_class, NULL,
+					MKDEV(MEM_MAJOR, devlist[i].minor),
 					NULL, devlist[i].name);
 
 	return 0;
--- gregkh-2.6.orig/drivers/char/misc.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/misc.c	2005-09-28 12:48:37.000000000 -0700
@@ -229,7 +229,7 @@
 		misc_minors[misc->minor >> 3] |= 1 << (misc->minor & 7);
 	dev = MKDEV(MISC_MAJOR, misc->minor);
 
-	misc->class = class_device_create(misc_class, dev, misc->dev,
+	misc->class = class_device_create(misc_class, NULL, dev, misc->dev,
 					  "%s", misc->name);
 	if (IS_ERR(misc->class)) {
 		err = PTR_ERR(misc->class);
--- gregkh-2.6.orig/drivers/char/ppdev.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/ppdev.c	2005-09-28 12:48:47.000000000 -0700
@@ -751,7 +751,7 @@
 
 static void pp_attach(struct parport *port)
 {
-	class_device_create(ppdev_class, MKDEV(PP_MAJOR, port->number),
+	class_device_create(ppdev_class, NULL, MKDEV(PP_MAJOR, port->number),
 			NULL, "parport%d", port->number);
 }
 
--- gregkh-2.6.orig/drivers/char/raw.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/raw.c	2005-09-28 12:49:01.000000000 -0700
@@ -127,7 +127,7 @@
 static void bind_device(struct raw_config_request *rq)
 {
 	class_device_destroy(raw_class, MKDEV(RAW_MAJOR, rq->raw_minor));
-	class_device_create(raw_class, MKDEV(RAW_MAJOR, rq->raw_minor),
+	class_device_create(raw_class, NULL, MKDEV(RAW_MAJOR, rq->raw_minor),
 				      NULL, "raw%d", rq->raw_minor);
 }
 
@@ -305,7 +305,7 @@
 		unregister_chrdev_region(dev, MAX_RAW_MINORS);
 		goto error;
 	}
-	class_device_create(raw_class, MKDEV(RAW_MAJOR, 0), NULL, "rawctl");
+	class_device_create(raw_class, NULL, MKDEV(RAW_MAJOR, 0), NULL, "rawctl");
 
 	return 0;
 
--- gregkh-2.6.orig/drivers/char/snsc.c	2005-08-28 16:41:01.000000000 -0700
+++ gregkh-2.6/drivers/char/snsc.c	2005-09-28 12:49:07.000000000 -0700
@@ -437,7 +437,7 @@
 				continue;
 			}
 
-			class_device_create(snsc_class, dev, NULL,
+			class_device_create(snsc_class, NULL, dev, NULL,
 						"%s", devname);
 
 			ia64_sn_irtr_intr_enable(scd->scd_nasid,
--- gregkh-2.6.orig/drivers/char/stallion.c	2005-09-27 03:04:16.000000000 -0700
+++ gregkh-2.6/drivers/char/stallion.c	2005-09-28 12:49:17.000000000 -0700
@@ -3087,7 +3087,7 @@
 
 	stallion_class = class_create(THIS_MODULE, "staliomem");
 	for (i = 0; i < 4; i++)
-		class_device_create(stallion_class, MKDEV(STL_SIOMEMMAJOR, i), NULL, "staliomem%d", i);
+		class_device_create(stallion_class, NULL, MKDEV(STL_SIOMEMMAJOR, i), NULL, "staliomem%d", i);
 
 	stl_serial->owner = THIS_MODULE;
 	stl_serial->driver_name = stl_drvname;
--- gregkh-2.6.orig/drivers/char/tipar.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/tipar.c	2005-09-28 12:49:24.000000000 -0700
@@ -435,7 +435,7 @@
 		goto out;
 	}
 
-	class_device_create(tipar_class, MKDEV(TIPAR_MAJOR,
+	class_device_create(tipar_class, NULL, MKDEV(TIPAR_MAJOR,
 			TIPAR_MINOR + nr), NULL, "par%d", nr);
 
 	/* Display informations */
--- gregkh-2.6.orig/drivers/char/tty_io.c	2005-09-27 03:04:16.000000000 -0700
+++ gregkh-2.6/drivers/char/tty_io.c	2005-09-28 12:49:49.000000000 -0700
@@ -2724,7 +2724,7 @@
 		pty_line_name(driver, index, name);
 	else
 		tty_line_name(driver, index, name);
-	class_device_create(tty_class, dev, device, name);
+	class_device_create(tty_class, NULL, dev, device, name);
 }
 
 /**
@@ -2977,20 +2977,20 @@
 	if (cdev_add(&tty_cdev, MKDEV(TTYAUX_MAJOR, 0), 1) ||
 	    register_chrdev_region(MKDEV(TTYAUX_MAJOR, 0), 1, "/dev/tty") < 0)
 		panic("Couldn't register /dev/tty driver\n");
-	class_device_create(tty_class, MKDEV(TTYAUX_MAJOR, 0), NULL, "tty");
+	class_device_create(tty_class, NULL, MKDEV(TTYAUX_MAJOR, 0), NULL, "tty");
 
 	cdev_init(&console_cdev, &console_fops);
 	if (cdev_add(&console_cdev, MKDEV(TTYAUX_MAJOR, 1), 1) ||
 	    register_chrdev_region(MKDEV(TTYAUX_MAJOR, 1), 1, "/dev/console") < 0)
 		panic("Couldn't register /dev/console driver\n");
-	class_device_create(tty_class, MKDEV(TTYAUX_MAJOR, 1), NULL, "console");
+	class_device_create(tty_class, NULL, MKDEV(TTYAUX_MAJOR, 1), NULL, "console");
 
 #ifdef CONFIG_UNIX98_PTYS
 	cdev_init(&ptmx_cdev, &ptmx_fops);
 	if (cdev_add(&ptmx_cdev, MKDEV(TTYAUX_MAJOR, 2), 1) ||
 	    register_chrdev_region(MKDEV(TTYAUX_MAJOR, 2), 1, "/dev/ptmx") < 0)
 		panic("Couldn't register /dev/ptmx driver\n");
-	class_device_create(tty_class, MKDEV(TTYAUX_MAJOR, 2), NULL, "ptmx");
+	class_device_create(tty_class, NULL, MKDEV(TTYAUX_MAJOR, 2), NULL, "ptmx");
 #endif
 
 #ifdef CONFIG_VT
@@ -2998,7 +2998,7 @@
 	if (cdev_add(&vc0_cdev, MKDEV(TTY_MAJOR, 0), 1) ||
 	    register_chrdev_region(MKDEV(TTY_MAJOR, 0), 1, "/dev/vc/0") < 0)
 		panic("Couldn't register /dev/tty0 driver\n");
-	class_device_create(tty_class, MKDEV(TTY_MAJOR, 0), NULL, "tty0");
+	class_device_create(tty_class, NULL, MKDEV(TTY_MAJOR, 0), NULL, "tty0");
 
 	vty_init();
 #endif
--- gregkh-2.6.orig/drivers/char/ftape/zftape/zftape-init.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/ftape/zftape/zftape-init.c	2005-09-28 12:51:32.000000000 -0700
@@ -330,12 +330,12 @@
 
 	zft_class = class_create(THIS_MODULE, "zft");
 	for (i = 0; i < 4; i++) {
-		class_device_create(zft_class, MKDEV(QIC117_TAPE_MAJOR, i), NULL, "qft%i", i);
-		class_device_create(zft_class, MKDEV(QIC117_TAPE_MAJOR, i + 4), NULL, "nqft%i", i);
-		class_device_create(zft_class, MKDEV(QIC117_TAPE_MAJOR, i + 16), NULL, "zqft%i", i);
-		class_device_create(zft_class, MKDEV(QIC117_TAPE_MAJOR, i + 20), NULL, "nzqft%i", i);
-		class_device_create(zft_class, MKDEV(QIC117_TAPE_MAJOR, i + 32), NULL, "rawqft%i", i);
-		class_device_create(zft_class, MKDEV(QIC117_TAPE_MAJOR, i + 36), NULL, "nrawrawqft%i", i);
+		class_device_create(zft_class, NULL, MKDEV(QIC117_TAPE_MAJOR, i), NULL, "qft%i", i);
+		class_device_create(zft_class, NULL, MKDEV(QIC117_TAPE_MAJOR, i + 4), NULL, "nqft%i", i);
+		class_device_create(zft_class, NULL, MKDEV(QIC117_TAPE_MAJOR, i + 16), NULL, "zqft%i", i);
+		class_device_create(zft_class, NULL, MKDEV(QIC117_TAPE_MAJOR, i + 20), NULL, "nzqft%i", i);
+		class_device_create(zft_class, NULL, MKDEV(QIC117_TAPE_MAJOR, i + 32), NULL, "rawqft%i", i);
+		class_device_create(zft_class, NULL, MKDEV(QIC117_TAPE_MAJOR, i + 36), NULL, "nrawrawqft%i", i);
 	}
 
 #ifdef CONFIG_ZFT_COMPRESSOR
--- gregkh-2.6.orig/drivers/char/ipmi/ipmi_devintf.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/ipmi/ipmi_devintf.c	2005-09-28 12:51:41.000000000 -0700
@@ -794,7 +794,7 @@
 {
 	dev_t dev = MKDEV(ipmi_major, if_num);
 
-	class_device_create(ipmi_class, dev, NULL, "ipmi%d", if_num);
+	class_device_create(ipmi_class, NULL, dev, NULL, "ipmi%d", if_num);
 }
 
 static void ipmi_smi_gone(int if_num)
--- gregkh-2.6.orig/drivers/char/vc_screen.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/vc_screen.c	2005-09-28 12:50:58.000000000 -0700
@@ -477,8 +477,8 @@
 
 void vcs_make_devfs(struct tty_struct *tty)
 {
-	class_device_create(vc_class, MKDEV(VCS_MAJOR, tty->index + 1), NULL, "vcs%u", tty->index + 1);
-	class_device_create(vc_class, MKDEV(VCS_MAJOR, tty->index + 129), NULL, "vcsa%u", tty->index + 1);
+	class_device_create(vc_class, NULL, MKDEV(VCS_MAJOR, tty->index + 1), NULL, "vcs%u", tty->index + 1);
+	class_device_create(vc_class, NULL, MKDEV(VCS_MAJOR, tty->index + 129), NULL, "vcsa%u", tty->index + 1);
 }
 void vcs_remove_devfs(struct tty_struct *tty)
 {
@@ -492,7 +492,7 @@
 		panic("unable to get major %d for vcs device", VCS_MAJOR);
 	vc_class = class_create(THIS_MODULE, "vc");
 
-	class_device_create(vc_class, MKDEV(VCS_MAJOR, 0), NULL, "vcs");
-	class_device_create(vc_class, MKDEV(VCS_MAJOR, 128), NULL, "vcsa");
+	class_device_create(vc_class, NULL, MKDEV(VCS_MAJOR, 0), NULL, "vcs");
+	class_device_create(vc_class, NULL, MKDEV(VCS_MAJOR, 128), NULL, "vcsa");
 	return 0;
 }
--- gregkh-2.6.orig/drivers/char/viotape.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/char/viotape.c	2005-09-28 12:51:11.000000000 -0700
@@ -954,9 +954,9 @@
 	state[i].cur_part = 0;
 	for (j = 0; j < MAX_PARTITIONS; ++j)
 		state[i].part_stat_rwi[j] = VIOT_IDLE;
-	class_device_create(tape_class, MKDEV(VIOTAPE_MAJOR, i), NULL,
+	class_device_create(tape_class, NULL, MKDEV(VIOTAPE_MAJOR, i), NULL,
 			"iseries!vt%d", i);
-	class_device_create(tape_class, MKDEV(VIOTAPE_MAJOR, i | 0x80),
+	class_device_create(tape_class, NULL, MKDEV(VIOTAPE_MAJOR, i | 0x80),
 			NULL, "iseries!nvt%d", i);
 	sprintf(tapename, "iseries/vt%d", i);
 	printk(VIOTAPE_KERN_INFO "tape %s is iSeries "
--- gregkh-2.6.orig/drivers/hwmon/hwmon.c	2005-09-20 08:14:19.000000000 -0700
+++ gregkh-2.6/drivers/hwmon/hwmon.c	2005-09-28 12:51:49.000000000 -0700
@@ -45,7 +45,7 @@
 		return ERR_PTR(-ENOMEM);
 
 	id = id & MAX_ID_MASK;
-	cdev = class_device_create(hwmon_class, MKDEV(0,0), dev,
+	cdev = class_device_create(hwmon_class, NULL, MKDEV(0,0), dev,
 					HWMON_ID_FORMAT, id);
 
 	if (IS_ERR(cdev))
--- gregkh-2.6.orig/drivers/ide/ide-tape.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/ide/ide-tape.c	2005-09-28 12:52:45.000000000 -0700
@@ -4880,9 +4880,9 @@
 
 	idetape_setup(drive, tape, minor);
 
-	class_device_create(idetape_sysfs_class,
+	class_device_create(idetape_sysfs_class, NULL,
 			MKDEV(IDETAPE_MAJOR, minor), dev, "%s", tape->name);
-	class_device_create(idetape_sysfs_class,
+	class_device_create(idetape_sysfs_class, NULL,
 			MKDEV(IDETAPE_MAJOR, minor + 128), dev, "n%s", tape->name);
 
 	g->fops = &idetape_block_ops;
--- gregkh-2.6.orig/drivers/ieee1394/dv1394.c	2005-09-27 03:04:14.000000000 -0700
+++ gregkh-2.6/drivers/ieee1394/dv1394.c	2005-09-28 12:52:53.000000000 -0700
@@ -2345,7 +2345,7 @@
 
 	ohci = (struct ti_ohci *)host->hostdata;
 
-	class_device_create(hpsb_protocol_class, MKDEV(
+	class_device_create(hpsb_protocol_class, NULL, MKDEV(
 		IEEE1394_MAJOR,	IEEE1394_MINOR_BLOCK_DV1394 * 16 + (id<<2)), 
 		NULL, "dv1394-%d", id);
 
--- gregkh-2.6.orig/drivers/ieee1394/raw1394.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/ieee1394/raw1394.c	2005-09-28 12:53:01.000000000 -0700
@@ -2903,7 +2903,7 @@
 
 	hpsb_register_highlevel(&raw1394_highlevel);
 
-	if (IS_ERR(class_device_create(hpsb_protocol_class, MKDEV(
+	if (IS_ERR(class_device_create(hpsb_protocol_class, NULL, MKDEV(
 		IEEE1394_MAJOR,	IEEE1394_MINOR_BLOCK_RAW1394 * 16), 
 		NULL, RAW1394_DEVICE_NAME))) {
 		ret = -EFAULT;
--- gregkh-2.6.orig/drivers/ieee1394/video1394.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/ieee1394/video1394.c	2005-09-28 12:53:08.000000000 -0700
@@ -1369,7 +1369,7 @@
 	hpsb_set_hostinfo_key(&video1394_highlevel, host, ohci->host->id);
 
 	minor = IEEE1394_MINOR_BLOCK_VIDEO1394 * 16 + ohci->host->id;
-	class_device_create(hpsb_protocol_class, MKDEV(
+	class_device_create(hpsb_protocol_class, NULL, MKDEV(
 		IEEE1394_MAJOR,	minor), 
 		NULL, "%s-%d", VIDEO1394_DRIVER_NAME, ohci->host->id);
 }
--- gregkh-2.6.orig/drivers/infiniband/core/ucm.c	2005-09-20 08:14:19.000000000 -0700
+++ gregkh-2.6/drivers/infiniband/core/ucm.c	2005-09-28 12:53:15.000000000 -0700
@@ -1300,7 +1300,7 @@
 		goto err_class;
 	}
 
-	class_device_create(ib_ucm_class, IB_UCM_DEV, NULL, "ucm");
+	class_device_create(ib_ucm_class, NULL, IB_UCM_DEV, NULL, "ucm");
 
 	idr_init(&ctx_id_table);
 	init_MUTEX(&ctx_id_mutex);
--- gregkh-2.6.orig/drivers/isdn/capi/capi.c	2005-09-27 03:04:16.000000000 -0700
+++ gregkh-2.6/drivers/isdn/capi/capi.c	2005-09-28 12:55:27.000000000 -0700
@@ -1503,7 +1503,7 @@
 		return PTR_ERR(capi_class);
 	}
 
-	class_device_create(capi_class, MKDEV(capi_major, 0), NULL, "capi");
+	class_device_create(capi_class, NULL, MKDEV(capi_major, 0), NULL, "capi");
 
 #ifdef CONFIG_ISDN_CAPI_MIDDLEWARE
 	if (capinc_tty_init() < 0) {
--- gregkh-2.6.orig/drivers/macintosh/adb.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/macintosh/adb.c	2005-09-28 12:55:34.000000000 -0700
@@ -902,5 +902,5 @@
 	adb_dev_class = class_create(THIS_MODULE, "adb");
 	if (IS_ERR(adb_dev_class))
 		return;
-	class_device_create(adb_dev_class, MKDEV(ADB_MAJOR, 0), NULL, "adb");
+	class_device_create(adb_dev_class, NULL, MKDEV(ADB_MAJOR, 0), NULL, "adb");
 }
--- gregkh-2.6.orig/drivers/media/dvb/dvb-core/dvbdev.c	2005-09-27 03:04:14.000000000 -0700
+++ gregkh-2.6/drivers/media/dvb/dvb-core/dvbdev.c	2005-09-28 12:55:41.000000000 -0700
@@ -231,7 +231,7 @@
 
 	list_add_tail (&dvbdev->list_head, &adap->device_list);
 
-	class_device_create(dvb_class, MKDEV(DVB_MAJOR, nums2minor(adap->num, type, id)),
+	class_device_create(dvb_class, NULL, MKDEV(DVB_MAJOR, nums2minor(adap->num, type, id)),
 			    NULL, "dvb%d.%s%d", adap->num, dnames[type], id);
 
 	dprintk("DVB: register adapter%d/%s%d @ minor: %i (0x%02x)\n",
--- gregkh-2.6.orig/drivers/message/i2o/iop.c	2005-09-27 03:01:35.000000000 -0700
+++ gregkh-2.6/drivers/message/i2o/iop.c	2005-09-28 12:55:58.000000000 -0700
@@ -1141,7 +1141,7 @@
 		goto iop_reset;
 	}
 
-	c->classdev = class_device_create(i2o_controller_class, 0,
+	c->classdev = class_device_create(i2o_controller_class, NULL, MKDEV(0,0),
 			&c->device, "iop%d", c->unit);
 	if (IS_ERR(c->classdev)) {
 		osm_err("%s: could not add controller class\n", c->name);
--- gregkh-2.6.orig/drivers/mtd/mtdchar.c	2005-08-28 16:41:01.000000000 -0700
+++ gregkh-2.6/drivers/mtd/mtdchar.c	2005-09-28 12:56:11.000000000 -0700
@@ -24,10 +24,10 @@
 	if (!mtd)
 		return;
 
-	class_device_create(mtd_class, MKDEV(MTD_CHAR_MAJOR, mtd->index*2),
+	class_device_create(mtd_class, NULL, MKDEV(MTD_CHAR_MAJOR, mtd->index*2),
 			    NULL, "mtd%d", mtd->index);
 	
-	class_device_create(mtd_class, 
+	class_device_create(mtd_class, NULL,
 			    MKDEV(MTD_CHAR_MAJOR, mtd->index*2+1),
 			    NULL, "mtd%dro", mtd->index);
 }
--- gregkh-2.6.orig/drivers/net/ppp_generic.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/net/ppp_generic.c	2005-09-28 12:56:17.000000000 -0700
@@ -862,7 +862,7 @@
 			err = PTR_ERR(ppp_class);
 			goto out_chrdev;
 		}
-		class_device_create(ppp_class, MKDEV(PPP_MAJOR, 0), NULL, "ppp");
+		class_device_create(ppp_class, NULL, MKDEV(PPP_MAJOR, 0), NULL, "ppp");
 	}
 
 out:
--- gregkh-2.6.orig/drivers/net/wan/cosa.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/net/wan/cosa.c	2005-09-28 12:56:22.000000000 -0700
@@ -398,7 +398,7 @@
 		goto out_chrdev;
 	}
 	for (i=0; i<nr_cards; i++) {
-		class_device_create(cosa_class, MKDEV(cosa_major, i),
+		class_device_create(cosa_class, NULL, MKDEV(cosa_major, i),
 				NULL, "cosa%d", i);
 	}
 	err = 0;
--- gregkh-2.6.orig/drivers/s390/char/tape_class.c	2005-08-28 16:41:01.000000000 -0700
+++ gregkh-2.6/drivers/s390/char/tape_class.c	2005-09-28 12:56:29.000000000 -0700
@@ -72,6 +72,7 @@
 
 	tcd->class_device = class_device_create(
 				tape_class,
+				NULL,
 				tcd->char_device->dev,
 				device,
 				"%s", tcd->device_name
--- gregkh-2.6.orig/drivers/s390/char/vmlogrdr.c	2005-08-28 16:41:01.000000000 -0700
+++ gregkh-2.6/drivers/s390/char/vmlogrdr.c	2005-09-28 12:56:34.000000000 -0700
@@ -787,6 +787,7 @@
 		return ret;
 	}
 	priv->class_device = class_device_create(
+				NULL,
 				vmlogrdr_class,
 				MKDEV(vmlogrdr_major, priv->minor_num),
 				dev,
--- gregkh-2.6.orig/drivers/scsi/ch.c	2005-09-20 08:14:27.000000000 -0700
+++ gregkh-2.6/drivers/scsi/ch.c	2005-09-28 12:56:39.000000000 -0700
@@ -936,7 +936,7 @@
 	if (init)
 		ch_init_elem(ch);
 
-	class_device_create(ch_sysfs_class,
+	class_device_create(ch_sysfs_class, NULL,
 			    MKDEV(SCSI_CHANGER_MAJOR,ch->minor),
 			    dev, "s%s", ch->name);
 
--- gregkh-2.6.orig/drivers/scsi/osst.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/scsi/osst.c	2005-09-28 12:56:46.000000000 -0700
@@ -5624,7 +5624,7 @@
 
 	if (!osst_sysfs_valid) return;
 
-	osst_class_member = class_device_create(osst_sysfs_class, dev, device, "%s", name);
+	osst_class_member = class_device_create(osst_sysfs_class, NULL, dev, device, "%s", name);
 	if (IS_ERR(osst_class_member)) {
 		printk(KERN_WARNING "osst :W: Unable to add sysfs class member %s\n", name);
 		return;
--- gregkh-2.6.orig/drivers/scsi/sg.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/scsi/sg.c	2005-09-28 12:57:11.000000000 -0700
@@ -1544,7 +1544,7 @@
 	if (sg_sysfs_valid) {
 		struct class_device * sg_class_member;
 
-		sg_class_member = class_device_create(sg_sysfs_class,
+		sg_class_member = class_device_create(sg_sysfs_class, NULL,
 				MKDEV(SCSI_GENERIC_MAJOR, k), 
 				cl_dev->dev, "%s", 
 				disk->disk_name);
--- gregkh-2.6.orig/drivers/scsi/st.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/scsi/st.c	2005-09-28 12:57:16.000000000 -0700
@@ -4356,7 +4356,7 @@
 		snprintf(name, 10, "%s%s%s", rew ? "n" : "",
 			 STp->disk->disk_name, st_formats[i]);
 		st_class_member =
-			class_device_create(st_sysfs_class,
+			class_device_create(st_sysfs_class, NULL,
 					    MKDEV(SCSI_TAPE_MAJOR,
 						  TAPE_MINOR(dev_num, mode, rew)),
 					    &STp->device->sdev_gendev, "%s", name);
--- gregkh-2.6.orig/drivers/input/evdev.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/input/evdev.c	2005-09-28 12:57:55.000000000 -0700
@@ -686,7 +686,7 @@
 
 	evdev_table[minor] = evdev;
 
-	class_device_create(input_class,
+	class_device_create(input_class, NULL,
 			MKDEV(INPUT_MAJOR, EVDEV_MINOR_BASE + minor),
 			dev->dev, "event%d", minor);
 
--- gregkh-2.6.orig/drivers/input/joydev.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/input/joydev.c	2005-09-28 12:57:59.000000000 -0700
@@ -513,7 +513,7 @@
 
 	joydev_table[minor] = joydev;
 
-	class_device_create(input_class,
+	class_device_create(input_class, NULL,
 			MKDEV(INPUT_MAJOR, JOYDEV_MINOR_BASE + minor),
 			dev->dev, "js%d", minor);
 
--- gregkh-2.6.orig/drivers/input/mousedev.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/input/mousedev.c	2005-09-28 12:58:13.000000000 -0700
@@ -648,7 +648,7 @@
 
 	mousedev_table[minor] = mousedev;
 
-	class_device_create(input_class,
+	class_device_create(input_class, NULL,
 			MKDEV(INPUT_MAJOR, MOUSEDEV_MINOR_BASE + minor),
 			dev->dev, "mouse%d", minor);
 
@@ -734,7 +734,7 @@
 	mousedev_mix.exist = 1;
 	mousedev_mix.minor = MOUSEDEV_MIX;
 
-	class_device_create(input_class,
+	class_device_create(input_class, NULL,
 			MKDEV(INPUT_MAJOR, MOUSEDEV_MINOR_BASE + MOUSEDEV_MIX), NULL, "mice");
 
 #ifdef CONFIG_INPUT_MOUSEDEV_PSAUX
--- gregkh-2.6.orig/drivers/input/tsdev.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/drivers/input/tsdev.c	2005-09-28 12:58:19.000000000 -0700
@@ -409,7 +409,7 @@
 
 	tsdev_table[minor] = tsdev;
 
-	class_device_create(input_class,
+	class_device_create(input_class, NULL,
 			MKDEV(INPUT_MAJOR, TSDEV_MINOR_BASE + minor),
 			dev->dev, "ts%d", minor);
 
--- gregkh-2.6.orig/arch/i386/kernel/cpuid.c	2005-08-28 16:41:01.000000000 -0700
+++ gregkh-2.6/arch/i386/kernel/cpuid.c	2005-09-28 13:33:09.000000000 -0700
@@ -163,7 +163,7 @@
 	int err = 0;
 	struct class_device *class_err;
 
-	class_err = class_device_create(cpuid_class, MKDEV(CPUID_MAJOR, i), NULL, "cpu%d",i);
+	class_err = class_device_create(cpuid_class, NULL, MKDEV(CPUID_MAJOR, i), NULL, "cpu%d",i);
 	if (IS_ERR(class_err))
 		err = PTR_ERR(class_err);
 	return err;
--- gregkh-2.6.orig/arch/i386/kernel/msr.c	2005-09-20 08:14:07.000000000 -0700
+++ gregkh-2.6/arch/i386/kernel/msr.c	2005-09-28 13:33:15.000000000 -0700
@@ -246,7 +246,7 @@
 	int err = 0;
 	struct class_device *class_err;
 
-	class_err = class_device_create(msr_class, MKDEV(MSR_MAJOR, i), NULL, "msr%d",i);
+	class_err = class_device_create(msr_class, NULL, MKDEV(MSR_MAJOR, i), NULL, "msr%d",i);
 	if (IS_ERR(class_err)) 
 		err = PTR_ERR(class_err);
 	return err;
--- gregkh-2.6.orig/sound/core/sound.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/sound/core/sound.c	2005-09-28 13:34:11.000000000 -0700
@@ -223,7 +223,7 @@
 	list_add_tail(&preg->list, &snd_minors_hash[SNDRV_MINOR_CARD(minor)]);
 	if (card)
 		device = card->dev;
-	class_device_create(sound_class, MKDEV(major, minor), device, "%s", name);
+	class_device_create(sound_class, NULL, MKDEV(major, minor), device, "%s", name);
 
 	up(&sound_mutex);
 	return 0;
--- gregkh-2.6.orig/sound/oss/soundcard.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/sound/oss/soundcard.c	2005-09-28 13:34:59.000000000 -0700
@@ -563,7 +563,7 @@
 	sound_dmap_flag = (dmabuf > 0 ? 1 : 0);
 
 	for (i = 0; i < sizeof (dev_list) / sizeof *dev_list; i++) {
-		class_device_create(sound_class,
+		class_device_create(sound_class, NULL,
 				    MKDEV(SOUND_MAJOR, dev_list[i].minor),
 				    NULL, "%s", dev_list[i].name);
 
@@ -571,7 +571,7 @@
 			continue;
 
 		for (j = 1; j < *dev_list[i].num; j++)
-			class_device_create(sound_class,
+			class_device_create(sound_class, NULL,
 					    MKDEV(SOUND_MAJOR, dev_list[i].minor + (j*0x10)),
 					    NULL, "%s%d", dev_list[i].name, j);
 	}
--- gregkh-2.6.orig/sound/sound_core.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/sound/sound_core.c	2005-09-28 13:35:12.000000000 -0700
@@ -171,8 +171,9 @@
 	else
 		sprintf(s->name, "sound/%s%d", name, r / SOUND_STEP);
 
-	class_device_create(sound_class, MKDEV(SOUND_MAJOR, s->unit_minor),
-			    dev, s->name+6);
+	class_device_create(sound_class, NULL,
+			    MKDEV(SOUND_MAJOR, s->unit_minor), dev,
+			    "%s", s->name+6);
 	return r;
 
  fail:
--- gregkh-2.6.orig/fs/coda/psdev.c	2005-09-27 03:04:15.000000000 -0700
+++ gregkh-2.6/fs/coda/psdev.c	2005-09-28 13:42:19.000000000 -0700
@@ -368,8 +368,8 @@
 		goto out_chrdev;
 	}		
 	for (i = 0; i < MAX_CODADEVS; i++)
-		class_device_create(coda_psdev_class, MKDEV(CODA_PSDEV_MAJOR,i),
-				NULL, "cfs%d", i);
+		class_device_create(coda_psdev_class, NULL,
+				MKDEV(CODA_PSDEV_MAJOR,i), NULL, "cfs%d", i);
 	coda_sysctl_init();
 	goto out;
 
